src/quicr_client_raw_session.cpp

```diff
@@ -364,39 +371,18 @@ QuicRClientRawSession::subscribe(
   transport->enqueue(transport_context_id, transport_stream_id, msg.take());
 }
 
-void
-QuicRClientRawSession::removeSubscribeState(
-  const quicr::Namespace& quicr_namespace,
-  const SubscribeResult::SubscribeStatus& reason)
-{
-  std::lock_guard<std::mutex> _(session_mutex);
-
-  if (!!subscribe_state.count(quicr_namespace)) {
-    subscribe_state.erase(quicr_namespace);
-  }
-
-  if (!!sub_delegates.count(quicr_namespace)) {
-    if (auto sub_delegate = sub_delegates[quicr_namespace].lock())
-      sub_delegate->onSubscriptionEnded(quicr_namespace, reason);
-
-    sub_delegates.erase(quicr_namespace);
-  }
-}
-
 void
 QuicRClientRawSession::unsubscribe(const quicr::Namespace& quicr_namespace,
                                    const std::string& /* origin_url */,
                                    const std::string& /* auth_token */)
 {
-  // The removal of the delegate is done on receive of subscription ended
-  std::lock_guard<std::mutex> lock(session_mutex);
-
   messages::MessageBuffer msg{};
   messages::Unsubscribe unsub{ 0x1, quicr_namespace };
   msg << unsub;
 
-  removeSubscribeState(quicr_namespace,
-                       SubscribeResult::SubscribeStatus::ConnectionClosed);
+  std::lock_guard<std::mutex> _(session_mutex);
```

Comment:
Why put the lock here, rather than removeSubscruption? There's no data to protect here, and this results in holding the lock longer than necessary as a call is then made into transport.